import java.util.Scanner;

class ExceptionHandler {

    void handleArithmeticException() {
        Scanner sc = new Scanner(System.in);
        while (true) {
            try {
                System.out.print("Enter two numbers: ");
                int num1 = sc.nextInt();
                int num2 = sc.nextInt();
                System.out.println("Division result: " + (num1 / num2));
                break; // Exit loop on successful division
            } catch (ArithmeticException e) {
                System.out.println("Cannot divide by zero. Please enter a valid denominator.");
            } catch (Exception e) {
                System.out.println("Invalid input. Please enter integers.");
                sc.nextLine(); // Clear the invalid input
            }
        }
    }

    void handleArrayOutOfBounds(int[] arr) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            try {
                System.out.print("Enter index to access: ");
                int index = sc.nextInt();
                System.out.println(arr[index] + " is present at index " + index);
                break; // Exit loop on successful access
            } catch (ArrayIndexOutOfBoundsException e) {
                System.out.println("Invalid index. Try again.");
            } catch (Exception e) {
                System.out.println("Invalid input. Please enter an integer.");
                sc.nextLine(); // Clear the invalid input
            }
        }
    }

    void handleNumberFormat() {
        Scanner sc = new Scanner(System.in);
        while (true) {
            try {
                System.out.print("Enter two numbers: ");
                int a = Integer.parseInt(sc.next());
                int b = Integer.parseInt(sc.next());
                System.out.println("Entered integers: " + a + ", " + b);
                break; // Exit loop on successful parsing
            } catch (NumberFormatException e) {
                System.out.println("Input is not an integer. Please try again.");
            }
        }
    }
}

// Main class
public class Main {
    public static void main(String[] args) {
        ExceptionHandler handler = new ExceptionHandler();
        Scanner sc = new Scanner(System.in);
        int choice;

        do {
            System.out.println("\nChoose an error type to check:");
            System.out.println("1] Arithmetic Exception");
            System.out.println("2] Array Out Of Bounds");
            System.out.println("3] Number Format");
            System.out.println("4] Exit");
            choice = sc.nextInt();

            switch (choice) {
                case 1:
                    handler.handleArithmeticException();
                    break;
                case 2:
                    handler.handleArrayOutOfBounds(new int[]{1, 4, 5, 2, 3, 8});
                    break;
                case 3:
                    handler.handleNumberFormat();
                    break;
                case 4:
                    System.out.println("Exiting...");
                    break;
                default:
                    System.out.println("Invalid choice!");
            }
        } while (choice != 4);
        
        sc.close();
    }
}
